lualib/resty/events/queue.lua

```diff
@@ -1,33 +1,41 @@
 local semaphore = require "ngx.semaphore"
 
+local table_new = require "table.new"
+
 local assert = assert
 local setmetatable = setmetatable
-local table_insert = table.insert
-local table_remove = table.remove
 
 
 local _M = {}
 local _MT = { __index = _M, }
 
-local MAX_QUEUE_LEN = 1024
+local DEFAULT_MAX_QUEUE_LEN = 1024 * 10
+local DEFAULT_QUEUE_LEN = 4096
 
-function _M.new()
+function _M.new(max_len)
     local self = {
         semaphore = assert(semaphore.new()),
-        count = 0,
+        max = max_len or DEFAULT_MAX_QUEUE_LEN,
+
+        elts = table_new(DEFAULT_QUEUE_LEN, 0),
```

Comment:
I think we need to restrict the upper bound on initial table size:

```suggestion
        elts = table_new(math.min(DEFAULT_QUEUE_LEN, max), 0),
```

Reply:
thanks, done.